src/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.Unix.cs

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.IO.Compression
+{
+    public partial class ZipArchiveEntry
+    {
+        internal const ZipVersionMadeByPlatform CurrentZipPlatform = ZipVersionMadeByPlatform.Unix;
+
+        /// <summary>
+        /// To get the file name of a ZipArchiveEntry, we should be parsing the FullName based
+        /// on the path specifications and requirements of the OS that ZipArchive was created on.
+        /// This method takes in a FullName and the platform of the ZipArchiveEntry and returns
+        /// the platform-correct file name.
+        /// </summary>
+        /// <returns></returns>
+        internal static string ParseFileName(string path, ZipVersionMadeByPlatform madeByPlatform)
+        {
+            // Validation checking is done based on current OS, not source OS.
+            PathInternal.CheckInvalidPathChars(path);
+
+            if (madeByPlatform == ZipVersionMadeByPlatform.Windows)
+            {
+                int length = path.Length;
+                for (int i = length; --i >= 0;)
+                {
+                    char ch = path[i];
+                    if (ch == '\\' || ch == '/' || ch == ':')
+                        return path.Substring(i + 1, length - i - 1);
```

Comment:
Yes, it can be.. I see no point in specifying the length if we're just going to be calculating it to the end of the string anyways.

I took this code from the [GetFileName method in System.Runtime.Extensions](https://github.com/dotnet/corefx/blob/master/src/System.Runtime.Extensions/src/System/IO/Path.cs)